ihmc-avatar-interfaces/src/main/java/us/ihmc/avatar/AvatarMPCWholeBodyControllerCoreThread.java

```diff
@@ -0,0 +1,291 @@
+import us.ihmc.commons.Conversions;
+import us.ihmc.communication.controllerAPI.StatusMessageOutputManager;
+import us.ihmc.graphicsDescription.yoGraphics.YoGraphicsListRegistry;
+import us.ihmc.humanoidRobotics.model.CenterOfPressureDataHolder;
+import us.ihmc.mecano.multiBodySystem.interfaces.JointBasics;
+import us.ihmc.robotModels.FullHumanoidRobotModel;
+import us.ihmc.robotics.robotController.ModularRobotController;
+import us.ihmc.robotics.sensors.CenterOfMassDataHolder;
+import us.ihmc.robotics.sensors.CenterOfMassDataHolderReadOnly;
+import us.ihmc.robotics.sensors.ForceSensorDataHolder;
+import us.ihmc.robotics.sensors.ForceSensorDataHolderReadOnly;
+import us.ihmc.robotics.time.ExecutionTimer;
+import us.ihmc.ros2.RealtimeROS2Node;
+import us.ihmc.scs2.definition.yoGraphic.YoGraphicDefinition;
+import us.ihmc.scs2.definition.yoGraphic.YoGraphicGroupDefinition;
+import us.ihmc.sensorProcessing.model.RobotMotionStatusHolder;
+import us.ihmc.sensorProcessing.outputData.JointDesiredOutputListBasics;
+import us.ihmc.sensorProcessing.parameters.HumanoidRobotSensorInformation;
+import us.ihmc.sensorProcessing.simulatedSensors.SensorDataContext;
+import us.ihmc.wholeBodyController.CenterOfMassCalibrationTool;
+import us.ihmc.wholeBodyController.DRCOutputProcessor;
+import us.ihmc.yoVariables.registry.YoRegistry;
+import us.ihmc.yoVariables.variable.YoBoolean;
+import us.ihmc.yoVariables.variable.YoDouble;
+import us.ihmc.yoVariables.variable.YoLong;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Supplier;
+
+import static us.ihmc.avatar.AvatarMPCControllerThread.createListOfJointsToIgnore;
+
+public class AvatarMPCWholeBodyControllerCoreThread implements AvatarControllerThreadInterface
+{
+   private static final boolean CREATE_COM_CALIBRATION_TOOL = false;
+
+   private final YoRegistry registry = new YoRegistry("WholeBodyControllerCoreThread");
+   private final YoDouble wholeBodyControllerCoreTime = new YoDouble("WholeBodyControllerCoreTime", registry);
+   private final FullHumanoidRobotModel controllerCoreFullRobotModel;
+   private final YoBoolean firstTick = new YoBoolean("FirstTick", registry);
+   private final YoLong timeStamp = new YoLong("TimeStampWholeBodyControllerCore", registry);
+   private final YoLong timeStampOffset = new YoLong("TimestampOffsetWholeBodyControllerCore", registry);
+   private final HumanoidRobotMPCContextData humanoidRobotContextData;
+   private final YoBoolean runWholeBodyControllerCore = new YoBoolean("RunWholeBodyControllerCore", registry);
+   private final YoGraphicsListRegistry yoGraphicsListRegistry = new YoGraphicsListRegistry();
+   private final List<Supplier<YoGraphicDefinition>> scs2YoGraphicHolders = new ArrayList<>();
+   private ModularRobotController wholeBodyControllerCoreCalculator = new ModularRobotController("WBCC");
+   private final ExecutionTimer wholeBodyControllerCoreThreadTimer;
```

Comment:
this is never used.

Reply:
I forgot to remove it when I checked the computation time.